java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -82,4 +116,44 @@ private static boolean isBigIntegerPotentiallyBiggerThanLong(NewClassTree newCla
       return true;
     }
   }
+
+  private JavaQuickFix computeQuickFix(NewClassTree newClassTree) {
+    String message = "";
+    JavaTextEdit textEdit = null;
+    String className = QuickFixHelper.contentForTree(newClassTree.identifier(), context);
+    if (EMPTY_STRING_MATCHER.matches(newClassTree)) {
+      message = formatQuickFixMessage(className, "an empty string \"\"");
+      textEdit = JavaTextEdit.replaceTree(newClassTree, "\"\"");
+    } else if (BIG_INT_MATCHER.matches(newClassTree)) {
+      String arg = getFirstArgumentAsString(newClassTree).replace("\"", "") + "L";
+      String replacement = String.format("BigInteger.valueOf(%s)", arg);
+      message = formatQuickFixMessage(className, "\"BigInteger.valueOf()\" static method");
+      textEdit = JavaTextEdit.replaceTree(newClassTree, replacement);
+    } else {
+      message = formatQuickFixMessage(className, "the " + classToLiteral.get(className) + " literal passed as parameter");
+      String replacement = getFirstArgumentAsString(newClassTree);
+      textEdit = JavaTextEdit.replaceTree(newClassTree, replacement);
+    }
+    return JavaQuickFix.newQuickFix(message).addTextEdit(textEdit).build();
+  }
+
+  private String formatQuickFixMessage(String constructor, String replacement) {
+    return String.format(QUICK_FIX_MESSAGE, constructor, replacement);
+  }
+
+  private String getFirstArgumentAsString(NewClassTree newClassTree) {
+    ExpressionTree expr = newClassTree.arguments().get(0);
+    if (expr instanceof LiteralTreeImpl) {
+      LiteralTreeImpl arg = (LiteralTreeImpl) expr;
+      return arg.value();
+    } else if (expr instanceof IdentifierTreeImpl) {
+      IdentifierTreeImpl arg = (IdentifierTreeImpl) expr;
+      return arg.name();
+    } else if (expr instanceof TypeCastExpressionTreeImpl){
+      return QuickFixHelper.contentForTree(expr, context);
+    }else {
+      return "";
```

Comment:
I'm really not sure about this.
From what I understand, this means you will probably end up with an empty replacement, and so it's not equivalent. Are you sure?

Since it is not covered, you have no idea what will happen here. This is not tested